tests/core/test_batch_definition.py

```diff
@@ -64,64 +79,15 @@ def test_get_batch_retrieves_only_batch(mocker: pytest_mock.MockFixture):
     mock_asset = mocker.Mock(spec=DataAsset)
     batch_definition.set_data_asset(mock_asset)
 
-    mock_get_batch_list_from_batch_request = mock_asset.get_batch_list_from_batch_request
-
     mock_batch = mocker.Mock(spec=Batch)
-    mock_get_batch_list_from_batch_request.return_value = [mock_batch]
+    mock_asset.get_batch.return_value = mock_batch
 
     # Act
     batch = batch_definition.get_batch()
 
     # Assert
     assert batch == mock_batch
-    mock_get_batch_list_from_batch_request.assert_called_once_with(
-        batch_definition.build_batch_request()
-    )
-
-
-@pytest.mark.unit
-def test_get_batch_retrieves_last_batch(mocker: pytest_mock.MockFixture):
```

Comment:
Is there another test that asserts we get the last batch when the batch_request is underspecified?

Reply:
Yeah, as far as I can tell, we should be good there. A lot of the existing tests that used to test `get_batch_list_from_batch_request`were updated to test this. And just to be sure I added a net new `test_get_batch_partitioner__sort_ascending_respected` test mostly because I wanted something targeted to prove I didn't mess up that request index piece you asked for a comment on.